### PR TITLE
Use OpenTelemetry with HTTP/gRPC exporters in apm-integration

### PR DESCRIPTION
**Pull request by tlrx**
This pull request adds support for HTPP/gRPC with the OpenTelemetry SDK in the `apm-integration` plugin. It adds the required security permissions to the plugin to make it work - we should try to reduce this list one day.

The plumbing to pass Elastic's Cloud APM server credentials is not fully there yet but the current `ApmIT` test can be executed with `-Dtests.apm.endpoint=https://... -Dtests.apm.token=ABC` and traces should be sent to Elastic APM.

_Works on my machine&trade;_

### CODE REVIEW

**Review by AthenaEryma — APPROVED** [review]
Not being at all familiar with the OpenTelemetry API, but having reviewed all the relevant PRs up until this point: LGTM.

File: x-pack/plugin/apm-integration/src/main/plugin-metadata/plugin-security.policy [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+grant {
+    permission java.lang.RuntimePermission "accessClassInPackage.sun.misc";
+    permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
+    permission java.lang.RuntimePermission "accessDeclaredMembers";
+    // required by io.grpc.internal.DnsNameResolver in grpc-core
+    permission java.net.NetPermission "getProxySelector";
```
Comment (ywangd) [inline x-pack/plugin/apm-integration/src/main/plugin-metadata/plugin-security.policy:14]:
```suggestion
    permission java.net.NetPermission "getProxySelector";
    permission java.lang.RuntimePermission "getClassLoader";
    permission java.net.SocketPermission "*", "connect,resolve";
```
With these two permissions, I am able to start ES and start sending data to the cloud APM server. Somehow they are not needed by `ApmIT`, but are necessary when actually running the node.

**Review by ywangd — APPROVED** [review]
LGTM

Just need two more security permissions for allowing exporting data to a Cloud APM server.
I can also confirm the `ApmIT` teset run successful on my machine.

**Review by SylvainJuge — APPROVED** [review]

File: x-pack/plugin/apm-integration/src/main/plugin-metadata/plugin-security.policy (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/apm-integration/src/main/plugin-metadata/plugin-security.policy:14]: > With these two permissions, I am able to start ES and start sending data to the cloud APM server. Somehow they are not needed by `ApmIT`, but are necessary when actually running the node.

Thanks for catching this Yang, I did not have the time yesterday to get that far :)

**Comment by tlrx** [comment]
Thanks everyone!

**Review by DaveCTurner — COMMENTED** [review]
Sorry for the late review, I was having too much fun playing around with it. LGTM2

**Pull request merged**